dlp/risk.py

```diff
@@ -0,0 +1,884 @@
+        topic_id: The name of the Pub/Sub topic to notify once the job
+            completes.
+        subscription_id: The name of the Pub/Sub subscription to use when
+            listening for job completion notifications.
+        timeout: The number of seconds to wait for a response from the API.
+
+    Returns:
+        None; the response from the API is printed to the terminal.
+    """
+
+    # Import the client library.
+    import google.cloud.dlp
+
+    # This sample additionally uses Cloud Pub/Sub to receive results from
+    # potentially long-running operations.
+    import google.cloud.pubsub
+
+    # This sample also uses threading.Event() to wait for the job to finish.
+    import threading
+
+    # Instantiate a client.
+    dlp = google.cloud.dlp.DlpServiceClient()
+
+    # Convert the project id into a full resource id.
+    parent = dlp.project_path(project)
+
+    # Location info of the BigQuery table.
+    source_table = {
+        'project_id': table_project_id,
+        'dataset_id': dataset_id,
+        'table_id': table_id
+    }
+
+    # Tell the API where to send a notification when the job is complete.
+    actions = [{
+        'pub_sub': {'topic': '{}/topics/{}'.format(parent, topic_id)}
+    }]
+
+    # Configure risk analysis job
+    # Give the name of the numeric column to compute risk metrics for
+    risk_job = {
+        'privacy_metric': {
+            'numerical_stats_config': {
+                'field': {
+                    'name': column_name
+                }
+            }
+        },
+        'source_table': source_table,
+        'actions': actions
+    }
+
+    # Call API to start risk analysis job
+    operation = dlp.create_dlp_job(parent, risk_job=risk_job)
+
+    # Create a Pub/Sub client and find the subscription. The subscription is
+    # expected to already be listening to the topic.
+    subscriber = google.cloud.pubsub.SubscriberClient()
+    subscription_path = subscriber.subscription_path(
+        project, subscription_id)
+    subscription = subscriber.subscribe(subscription_path)
+
+    # Set up a callback to acknowledge a message. This closes around an event
+    # so that it can signal that it is done and the main thread can continue.
+    job_done = threading.Event()
+
+    def callback(message):
+        try:
+            if (message.attributes['DlpJobName'] == operation.name):
+                # This is the message we're looking for, so acknowledge it.
+                message.ack()
+
+                # Now that the job is done, fetch the results and print them.
+                job = dlp.get_dlp_job(operation.name)
+                results = job.risk_details.numerical_stats_result
+                print('Value Range: [{}, {}]'.format(
+                    results.min_value.integer_value,
+                    results.max_value.integer_value))
+                print('25% quantile value: {}\n'  # NOTE: change this to unique
```

Comment:
What does this #NOTE mean?

Reply:
This is a note to ask you about switching this to be more like the NodeJS samples. The NodeJS samples only print the quantiles with unique values. Here I am summarizing by providing the quartiles.

Reply:
I don't have a strong opinion on the output, since I don't really understand the functionality of the endpoint very well. = ) It should be fine as it is.